discord/ext/commands/core.py

```diff
@@ -256,7 +257,12 @@ def __init__(self, func, **kwargs):
         except AttributeError:
             cooldown = kwargs.get('cooldown')
         finally:
-            self._buckets = CooldownMapping(cooldown)
+            if cooldown is None:
+                self._buckets = CooldownMapping(cooldown, BucketType.default)
+            elif callable(cooldown[0]):
+                self._buckets = DynamicCooldownMapping(*cooldown)
+            else:
+                self._buckets = CooldownMapping(*cooldown)
```

Comment:
This doesn't work if `cooldown` isn't a tuple. However, there are no cases where this is true now, so probably remove it.